filebeat/tests/integration/filestream_test.go

```diff
@@ -505,11 +508,96 @@ logging:
 	requireRegistryEntryRemoved(t, workDir, "native")
 }
 
+// TestFilestreamCanMigrateID ensures the Filestream input can migrate state
+// when its ID is changed.
+// The way this test works is:
+//   - It uses a set of known files (testdata/migrate-id/) so the registry keys
+//     are known in advance because fingerprint is used
+//   - We ran Filebeat to populate the registry/ingest the files
+//   - We stop Filebeat and update the configuration
+//   - We ran Filebeat again and assert: no new data is ingested, all registry
+//     keys are correctly updated
+//
+// Filenames and their fingerprints:
+//   - 01.log: 6fb3cb6c565bdba1354f64a42dd47ef937964019400dd571f25c2cd13a9fb5be
+//   - 02.log: db8399294e69089070405b13d4f057672f3852fa8e0f56ce4b6c92398aef1b6a
+//   - 01.txt: 9ef9433360a276b14e8eae3864594c0108042c0828d3504b34c082dfc1cd43da
+//   - 02.txt: 10c6577e45f2b06631e11285210d8bd967ebf9786cf81ccfc9fef64bc01725cd
+func TestFilestreamCanMigrateID(t *testing.T) {
+	oldID := "first-id"
+	newID := "second-id"
+
+	testDataPath, err := filepath.Abs("./testdata")
+	if err != nil {
+		t.Fatalf("cannot get absolute path for 'testdata': %s", err)
+	}
+
+	// Get the absolute path for all files Filebeat will ingest
+	logFiles := []string{}
+	for _, f := range []string{"01.log", "02.log", "01.txt", "02.txt"} {
+		logFiles = append(logFiles, filepath.Join(testDataPath, "update-filestream-id", f))
+	}
+
+	filebeat := integration.NewBeat(
+		t,
+		"filebeat",
+		"../../filebeat.test",
+	)
+	workDir := filebeat.TempDir()
+	outputFile := filepath.Join(workDir, "output-file*")
+
+	vars := map[string]string{
+		"inputID":  oldID,
+		"homePath": workDir,
+		"testdata": testDataPath,
+	}
+	cfgYAML := getMigrateIDConfig(t, vars)
+	filebeat.WriteConfigFile(cfgYAML)
+	filebeat.Start()
+
+	// Wait for the file to be fully ingested
+	// eofMsg := fmt.Sprintf("End of file reached: %s; Backoff now.", logFilepath)
+	// filebeat.WaitForLogs(eofMsg, time.Second*10, "EOF was not reached")
```

Comment:
leftover from a refactor or debug?

Reply:
Yes, left over from a refactoring. I fixed it on e6f7d8cf1a